pyi.py

```diff
@@ -972,6 +1052,103 @@ def visit_ClassDef(self, node: ast.ClassDef) -> None:
             ):
                 self.error(statement, Y013)
 
+    def _check_exit_method(  # noqa: C901
```

Comment:
On the one hand, I agree with flake8 that this is quite complicated. On the other hand, it's good because all exit method checking is in the same place.

Reply:
Yes. I tried splitting it up, and, in my opinion, it made the code much less readable.